### ISSUE TITLE
Picking random colors can result in an IndexError

### ISSUE THREAD

**Issue by ri0t** [secretBiology/SecretColors#2]
### What i am trying to do
I am trying to pick a random color via `SecretColors.Palette(palette_name, show_warning=False).random()`

### What i expect to happen
A random color should be returned, most of the time, it does correctly do that.

### What happens instead
Apparently - very rarely - this happens:
```
  File "/home/riot/src/isomer/isomer_master/isomer/misc/__init__.py", line 906, in std_color
    color = SecretColors.Palette(palette_name, show_warning=False).random()
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/palette.py", line 368, in random
    force_gray=force_gray), alpha=alpha,
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/palette.py", line 297, in __random
    ending_shade=end_shade)
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/palette.py", line 274, in __random_pick
    no_of_colors)
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/__color.py", line 206, in random_between
    return [self.shade(x) for x in random_shades]
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/__color.py", line 206, in <listcomp>
    return [self.shade(x) for x in random_shades]
  File "/home/riot/.virtualenvs/isomer/lib/python3.7/site-packages/SecretColors/__color.py", line 186, in shade
    return col2[ind2]
IndexError: list index out of range
```

### Steps to Reproduce
```
import SecretColors as sc

for x in range(10000):
    _ = sc.Palette().random()
```

**Comment by rohitsuratekar** [comment]
Thank you @ri0t for pointing this out. I just checked, this is happening because there is bug in rounding off the lower numbers when you are drawing random numbers. I am planning on releasing major update to the library in next few days. It will include some major improvements and I will be able to fix this issue. For now, if you want workaround, you can set the `starting_shade` to 10. 

```
   import SecretColors as sc
    for x in range(10000):
        _ = sc.Palette().random(starting_shade=10)
```

**Comment by ri0t** [comment]
Uh, nice! I added an exception handler around it for now. But knowing you'll include a fix, i'll just git revert that here and wait for your patch, as i'm planning to release another minor soon, too.

Thanks a lot, SecretColors is really neat!

**Comment by rohitsuratekar** [comment]
Thanks. I am glad that at least someone (other than me) is using it :smile:

**Comment by rohitsuratekar** [comment]
Bug is fixed in `v1.2.0` and there are many new features to play around. Take a look at full [documentation](https://secretcolors.readthedocs.io/en/latest/) for the details. Please update to latest version with  `pip install SecretColors --upgrade`
